int32 id
Point[] corners
float64 z_rotation